### PR TITLE
Fix error message and improve error handling when loading file for Win32

### PR DESCRIPTION
**Pull request by bghgary**

### CODE REVIEW

File: Source/UrlRequest_Win32.cpp [hunk 1]
```diff
@@ -68,36 +68,41 @@ namespace UrlLib
...
                 return arcana::make_task(arcana::threadpool_scheduler, m_cancellationSource, [this, path] {
-                    std::ifstream file(path, std::ios::binary);
-                    if (!file.good())
+                    try
                     {
-                        std::stringstream msg;
-                        msg << "Failed to read file " << path.c_str();
+                        std::ifstream file{path, std::ios::binary | std::ios::ate};
+                        file.exceptions(std::ifstream::failbit | std::ifstream::badbit);
+                        m_fileResponseBuffer.resize(file.tellg());
```
Comment (docEdub) [inline Source/UrlRequest_Win32.cpp:95]: Why the change to `file.tellg()` from `std::filesystem::file_size(path)`?

**Review by docEdub — APPROVED** [review]

**Comment by ryantrem** [comment]
Why is it preferable to catch a specific error and then throw a generic error?

**Comment by bghgary** [comment]
> Why is it preferable to catch a specific error and then throw a generic error?

If the specific error had good information, I would have, but sadly it does not.

File: Source/UrlRequest_Win32.cpp (hunk 1 above)
Comment (bghgary) [reply Source/UrlRequest_Win32.cpp:95]: Extra file open for no reason.

**Review by ryantrem — APPROVED** [review]

File: Source/UrlRequest_Win32.cpp [hunk 2]
```diff
@@ -68,36 +68,41 @@ namespace UrlLib
             case UrlResponseType::String:
             {
                 return arcana::make_task(arcana::threadpool_scheduler, m_cancellationSource, [this, path] {
-                    std::ifstream file(path);
-                    if (!file.good())
+                    try
                     {
-                        std::stringstream msg;
-                        msg << "Failed to read file " << path.c_str();
```
Comment (bghgary) [inline Source/UrlRequest_Win32.cpp:75]: This is the main thing that was broken. This will result in a pointer being converted to a hex string.
